### PR TITLE
Remove `MoneyRequestConfirmationList` and replace to `MoneyTemporaryForRefactorRequestConfirmationList`

### PR DESCRIPTION
**Pull request by brunovjk**
### Details

This is a part of https://github.com/Expensify/App/issues/29107 and _is all about removing the original components and any of the duplicated efforts._

Old Component: `MoneyRequestConfirmationList`.
New Component `MoneyTemporaryForRefactorRequestConfirmationList`.
### Fixed Issues

$ https://github.com/Expensify/App/issues/39559
PROPOSAL: https://github.com/Expensify/App/issues/39559#issuecomment-2035695512

### Tests

This is a refactoring PR, everything should work the same as before.
#### The `MoneyRequestConfirmationList` component is utilized across various IOU types (send, split, request, track-expense) to confirm requests (manual, scan, distance). Therefore, comprehensive testing is required across all flows.

1. On the `Confirmation Page`, thoroughly test each field:
   - Verify that the initial value matches the data entered in previous steps (e.g., creating the request).
   - Edit each field, then press `Next/Request/Send/Track, ...` to proceed.
   - After proceeding, verify that the value remains consistent with the previous step.
   - Edit each field again and press `Next/Request/Send/Track, ...` without changing the value.
   - Verify that the value remains unchanged and all other fields are correct.
   - Edit each field again and press the `Back button` and confirm that the value remains unchanged and all other fields are correct.
   - On the `Confirmation Page` press the `Back button`, press `Next` from the previous page, and confirm that the `Confirmation Page` retains its state.

2. After completing the field testing, press the `Request` button and confirm redirection to the corresponding chat (report).

3. For Split flows, click on the newly created report preview and verify redirection to the Confirmation Page. Ensure all fields are correct and expected. Additionally, verify the ability to edit editable fields.

#### Below are steps to reach the Confirmation Page in each flow:

- **Send Money (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Send Money".
  3. Fill in the amount and press "Next".
  4. Add a contact/user.
  5. Verify the Confirmation Page and test all fields and buttons.
  6. Press "Pay elsewhere".
  7. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Manual (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Request Money".
  3. Click on "Manual" at the top.
  4. Fill in the amount and press "Next".
  5. Add a contact/user.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Scan (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Request Money".
  3. Click on "Scan" at the top.
  4. Upload a receipt.
  5. Add a contact/user.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Distance (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Request Money".
  3. Click on "Distance" at the top.
  4. Add two or more valid waypoints.
  5. Press "Next".
  6. Add a contact/user.
  7. Verify the Confirmation Page and test all fields and buttons.
  8. Press "Request".
  9. Confirm redirection to the report chat with the new created report preview.

- **Track Expense - Manual (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Track Expense".
  3. Click on "Manual" at the top.
  4. Fill in the amount and press "Next".
  5. Verify the Confirmation Page and test all fields and buttons.
  6. Press "Track Expense".
  7. Confirm redirection to the report chat with the new created report preview.

- **Track Expense - Scan (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Track Expense".
  3. Click on "Scan" at the top.
  4. Upload a receipt.
  5. Verify the Confirmation Page and test all fields and buttons.
  6. Press "Track Expense".
  7. Confirm redirection to the report chat with the new created report preview.

- **Track Expense - Distance (Global Fab)**:
  1. Click on the Global Fab.
  2. Press "Track Expense".
  3. Click on "Distance" at the top.
  4. Add two or more valid waypoints.
  5. Press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Track Expense".
  8. Confirm redirection to the report chat with the new created report preview.

- **Split Bill - Manual (Workspace Default Chat)**:
  1. Go to a workspace default chat (admins or announce rooms).
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Split Preview".
  10. Verify all fields have expected values.

- **Split Bill - Scan (Workspace Default Chat)**:
  1. Go to a workspace default chat (admins or announce rooms).
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Receipt Preview".
  10. Verify all fields have expected values (amount usually remains 0.00 awaiting scanning).
  11. Test all enable fields to ensure editing functionality.

- **Send Money (1:1 Chat)**:
  1. Go to a 1:1 chat.
  2. Click on the "+" icon.
  3. Press "Send Money".
  4. Fill in the amount and press "Next".
  5. Verify the Confirmation Page and test all fields and buttons.
  6. Press "Pay elsewhere".
  7. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Manual (1:1 Chat)**:
  1. Go to a 1:1 chat.
  2. Click on the "+" icon.
  3. Press "Request Money".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Scan (1:1 Chat)**:
  1. Go to a 1:1 chat.
  2. Click on the "+" icon.
  3. Press "Request Money".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Split Bill - Manual (1:1 Chat)**:
  1. Go to a 1:1 chat.
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Split Preview".
  10. Verify all fields have expected values.

- **Split Bill - Scan (1:1 Chat)**:
  1. Go to a 1:1 chat.
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Receipt Preview".
  10. Verify all fields have expected values (amount usually remains 0.00 awaiting scanning).
  11. Test all enable fields to ensure editing functionality.

- **Track Expense - Manual (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Track Expense".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Track expense".
  8. Confirm redirection to the report chat with the new created report preview.

- **Track Expense - Scan (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Track Expense".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Track expense".
  8. Confirm redirection to the report chat with the new created report preview.

- **Track Expense - Distance (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Track Expense".
  4. Click on "Distance" at the top.
  5. Add two or more valid waypoints.
  6. Press "Next".
  7. Verify the Confirmation Page and test all fields and buttons.
  8. Press "Track expense".
  9. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Manual (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Request Money".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Scan (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Request Money".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Request".
  8. Confirm redirection to the report chat with the new created report preview.

- **Request Money - Distance (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Request Money".
  4. Click on "Distance" at the top.
  5. Add two or more valid waypoints.
  6. Press "Next".
  7. Verify the Confirmation Page and test all fields and buttons.
  8. Press "Request".
  9. Confirm redirection to the report chat with the new created report preview.

- **Split Bill - Manual (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Manual" at the top.
  5. Fill in the amount and press "Next".
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Split Preview".
  10. Verify all fields have expected values.
  11. Test all enable fields to ensure editing functionality.

- **Split Bill - Scan (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Scan" at the top.
  5. Upload a receipt.
  6. Verify the Confirmation Page and test all fields and buttons.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Receipt Preview".
  10. Verify all fields have expected values (amount usually remains 0.00 awaiting scanning).
  11. Test all enable fields to ensure editing functionality.

- **Split Bill - Distance (Workspace Chat)**:
  1. Go to a workspace chat.
  2. Click on the "+" icon.
  3. Press "Split Bill".
  4. Click on "Distance" at the top.
  4. Add two or more valid waypoints.
  7. Press "Split".
  8. Confirm redirection to the report chat with the new created report preview.
  9. Click on the newly created "Receipt Preview".
  10. Verify all fields have expected values.
  11. Test all enable fields to ensure editing functionality.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as `Tests`, mirroring online ones with optimistic values.
- [x] Verify that no errors appear in the JS console
### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/95647348/e3d64efa-286e-4950-a94d-7cd091415f82

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/95647348/9ca80645-efc7-4fd9-bfe9-0e62778b756d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/95647348/3271df6f-87ba-4c43-8930-a8145d33a829

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/95647348/f7269996-5d88-4f78-bd41-4bb9ff8ee74e

</details>
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

### CODE REVIEW

**Comment by brunovjk** [comment]
@eh2077 our PR is ready for review. Im struggling with my Android build. Can you assist help me with tests/videos on Android? Thank you very much.

**Comment by brunovjk** [comment]
@eh2077 our PR is ready for review. Im struggling with my Android build. Can you assist me here with the videos? Thank you very much.

**Comment by brunovjk** [comment]
@eh2077 our PR is ready for review.
Im struggling with my Android build, can you assist me with test/videos here?
Thank you very much.

**Comment by eh2077** [comment]
@brunovjk I think we should test each IOU flow separately to ensure that these critical flows are not broken. 

We have four IOU types, [`send`, `split`, `request`, `track-expense`], and three request types, [`manual`, `scan`, `distance`]. There're 10 flows in total
1. `send` money - `manual`
2. `request` money - `manual`
3. `request` money - `scan`
4. `request` money - `distance` in workspace
5. `split` money - `manual`
6. `split` money - `scan`
7. `split` money - `distance` in workspace
8. `track-expense` - `manual`
9. `track-expense` - `scan`
10. `track-expense` - `distance`

Would you like to update test steps?

File: src/components/MoneyRequestConfirmationList.tsx [hunk 1]
```diff
@@ -462,24 +509,33 @@ function MoneyRequestConfirmationList({
...
+     * @param {String} paymentMethod
+     */
     const confirm = useCallback(
-        (paymentMethod: IouType | PaymentMethodType | undefined) => {
-            if (!selectedParticipants.length) {
+        (paymentMethod: PaymentMethodType | undefined) => {
+            if (selectedParticipants.length === 0) {
                 return;
             }
-            if (iouCategory.length > CONST.API_TRANSACTION_CATEGORY_MAX_LENGTH) {
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/components/MoneyRequestConfirmationList.tsx:479]: Yes! You are sure. In fact @eh2077 I had failed to check the history, I was missing some changes to migrate, including the one mentioned by you. I looked again, changed what was necessary and I believe we are good now. Thank you very much for the review.

**Comment by brunovjk** [comment]
I tested it on the 10 flows you mentioned and everything works fine (creating and editing when enable), we have to test everything carefully, all the fields, going back and forth,... . I only tested it on the `Browser` today, tomorrow I will come back and test it on the other platforms. Thank you very much for the review.

**Comment by brunovjk** [comment]
@eh2077 I changed what you suggested, tested it on those 10 flows you mentioned and everything works fine **now** (creating and editing when enable).
I only tested it on the `Browser` today, tomorrow I will come back and test it on the other platforms. I will redo the `Test` steps and videos.
Thank you very much for the review.

File: src/components/MoneyRequestConfirmationList.tsx [hunk 2]
```diff
@@ -294,40 +285,67 @@ function MoneyRequestConfirmationList({
...
 
     useEffect(() => {
         if (!shouldCalculateDistanceAmount) {
             return;
         }
 
         const amount = DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate ?? 0);
-        IOU.setMoneyRequestAmount(amount);
-    }, [shouldCalculateDistanceAmount, distance, rate, unit]);
+        IOU.setMoneyRequestAmount_temporaryForRefactor(transaction?.transactionID ?? '', amount, currency ?? '');
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/components/MoneyRequestConfirmationList.tsx]: I believe so, however, you were right about `SplitBill`, I changed it to be dynamic when going to `CREATE` OR `EDIT`. Can you take a look, please? Especially this [commit](https://github.com/Expensify/App/pull/40176/commits/6e0a24b0fb0f0633ae5ac725646e4c2c44089880)

**Comment by eh2077** [comment]
@brunovjk Cool! Can you help update the those test steps also? So that I can also test them.

**Comment by brunovjk** [comment]
> Can you help update the those test steps also? So that I can also test them.

Sure! All steps we found necessary are covered. They're a bit extensive now. If you find any unnecessary or redundant test cases, please let me know, and I'll make the adjustments. Thank you @eh2077.

**Comment by eh2077** [comment]
@brunovjk Thank you! Let me take some time to look into the code and test those flows.

**Comment by brunovjk** [comment]
> Thank you! Let me take some time to look into the code and test those flows.

Awesome!!! Im here if you need anything. Have a good work.

**Comment by eh2077** [comment]
I compared following flows with production and confirmed they all work as expected
1. Send Money (Global Fab)
2. Request Money - Manual (Global Fab)
3. Request Money - Scan (Global Fab)
4. Request Money - Distance (Global Fab)
5. Track Expense - Manual (Global Fab)
6. Track Expense - Scan (Global Fab)
7. Track Expense - Distance (Global Fab)
    Note: This feature is only available on main now
8. Split Bill - Manual (Workspace Default Chat)
9. Split Bill - Scan (Workspace Default Chat)
10. Split Bill - Distance (Workspace Default Chat)

**Comment by brunovjk** [comment]
Thank you very much @eh2077, what would be the next step here?

File: src/libs/actions/IOU.ts [hunk 3]
```diff
@@ -6020,7 +6016,6 @@ export {
     resetMoneyRequestInfo,
     clearMoneyRequest,
     updateMoneyRequestTypeParams,
-    setMoneyRequestAmount_temporaryForRefactor,
     setMoneyRequestBillable_temporaryForRefactor,
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/actions/IOU.ts:6019]: Hm, it's possible that some of these got missed in some of the issues, but once all these components are done being migrated, I still need to go back and scan through all the code to see what is left to cleanup. These `_temporaryForRefactor` are one of the items that was on that list to look for. I think it's OK to leave the other methods alone for now.

**Comment by eh2077** [comment]
@brunovjk I found a bug - [**Request Money - Scan**] uploading PDF receipt not working


https://github.com/Expensify/App/assets/117511920/8501d324-a063-4f71-9c6d-d4984e958d66

PDF files for testing

- [sample-protected.pdf](https://github.com/Expensify/App/files/15012943/sample-protected.pdf)
- [sample.pdf](https://github.com/Expensify/App/files/15012944/sample.pdf)

**Comment by brunovjk** [comment]
I still haven't found the cause, I reported it on [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1713382047884229). I'll come back tomorrow and investigate further.

**Comment by eh2077** [comment]
@brunovjk Thanks for starting this [discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1713382047884229) on Slack! The fix PR https://github.com/Expensify/App/pull/40385 should be merged soon.

**Comment by eh2077** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by brunovjk** [comment]
Nice!!! Thank you :D

**Review by eh2077 — APPROVED** [review]
LGTM and tested well!

**Comment by tgolen** [comment]
I can review this today, but there are a few conflicts currently. Can you please fix them before I review it? Thanks!

**Comment by brunovjk** [comment]
Resolved the conflicts, let me test again.

**Review by tgolen — APPROVED** [review]

**Comment by tgolen** [comment]
OK, the changes look fine to me. Let me know once you've run through a few tests and then I'll merge it.

**Comment by brunovjk** [comment]
Tested: `send`, `split`, `request`, `track-expense` using `manual`, `scan` and `distance`
Everything looking good :D

File: src/libs/actions/IOU.ts [hunk 4]
```diff
@@ -5820,10 +5820,6 @@ function setMoneyRequestId(id: string) {
     Onyx.merge(ONYXKEYS.IOU, {id});
 }
 
-function setMoneyRequestAmount(amount: number) {
-    Onyx.merge(ONYXKEYS.IOU, {amount});
-}
-
 function setMoneyRequestCurrency(currency: string) {
```
[the remark this answers is not in the document]
Comment (brunovjk) [reply src/libs/actions/IOU.ts:5823]: Thanks, let me check in a moment

[the remark this answers is not in the document]
Comment (brunovjk) [reply src/libs/actions/IOU.ts:5823]: > @brunovjk You left a zombie function. Nothing uses this function.

Sorry, but I don't understand, what do you mean by `Nothing uses this function.`?
https://github.com/search?q=repo%3AExpensify%2FApp%20setMoneyRequestAmount&type=code

Take a look at this https://github.com/Expensify/App/pull/40176#discussion_r1568961142, is that what you're talking about?
Thank you: D

[the remark this answers is not in the document]
Comment (brunovjk) [reply src/libs/actions/IOU.ts:5823]: Thank you :D

**Comment by brunovjk** [comment]
@Krishna2323 Should we remove only `transaction, transactionID` from this line:
https://github.com/Expensify/App/blob/035adb890e559f287368b75f8cbe138c20e3f73a/src/components/MoneyRequestConfirmationList.tsx#L523
Correct?

**Comment by brunovjk** [comment]
Thanks, @Krishna2323, for catching that. My apologies for the reintroduced issue. I'm available to submit another PR if the team agrees. Do we only need to remove `transaction` and `transactionID` 
https://github.com/Expensify/App/blob/035adb890e559f287368b75f8cbe138c20e3f73a/src/components/MoneyRequestConfirmationList.tsx#L523
and add back "eslint-disable-next-line"?

**Comment by brunovjk** [comment]
@eh2077 Actually not exactly like the OP of the [issue](https://github.com/Expensify/App/issues/38744). However, I can see some inconsistencies (different lag/delay time) when selecting and deselecting categories and tags. But absolutely nothing like the problem presented before.

**Pull request merged**